### PR TITLE
Feat/contract call tx page

### PR DESCRIPTION
**Pull request by aulneau**

### CODE REVIEW

File: src/components/token-transfer.tsx [hunk 1]
```diff
@@ -137,6 +138,20 @@ const ItemIcon = ({ type }: { type: EventSingle['event_type'] }) => {
       );
   }
 };
+
+const ItemThing = ({ data }: { data: any }) => {
+  if (data.contract_log) {
+    let value = data.contract_log.value;
+    const thing = deserializeCV(Buffer.from(data.contract_log.value.replace('0x', ''), 'hex'));
+
+    if (thing.type === 5 && 'address' in thing) {
```
Comment (kyranjamie) [inline src/components/token-transfer.tsx:147]: can you make a helper fn that describes what this does?

File: src/store/transactions/selectors.ts [hunk 2]
```diff
@@ -19,3 +20,15 @@ export const selectTransactionLastTxId = createSelector(
   selectTransactionsSlice,
   state => state.lastTxId
 );
+
+export const selectOriginContractSource = (contractName: string) =>
+  createSelector(selectTransactionsSlice, state => {
+    const address = contractName.split('.')[0];
+    const txs = Object.values(state.entities) as Transaction[];
```
Comment (kyranjamie) [inline src/store/transactions/selectors.ts:27]: Would be nice to use `.values` but we can avoid the typecasting doing this, think it's worthwhile as its more type-safe.

```suggestion
    const txs = Object.keys(state.entities).map(tx_id => state.entities[tx_id]);
```

File: src/store/transactions/selectors.ts [hunk 3]
```diff
@@ -19,3 +20,15 @@ export const selectTransactionLastTxId = createSelector(
...
+
+    const contract = txs.find(
+      (tx: Transaction) => tx.tx_type === 'smart_contract' && tx.sender_address === address
+    ) as SmartContractTransaction;
```
Comment (kyranjamie) [inline src/store/transactions/selectors.ts:31]: Should now be `SmartContractTransaction` implictly.

```suggestion
    const contract = txs.find(
      (tx: Transaction) => tx.tx_type === 'smart_contract' && tx.sender_address === address
    );
```

File: src/components/token-transfer.tsx (hunk 1 above)
Comment (aulneau) [reply src/components/token-transfer.tsx:147]: Yeah def should have renamed this lol

**Pull request merged**